programs/stake/src/stake_state.rs

```diff
@@ -303,7 +324,18 @@ fn calculate_stake_points_and_credits(
         }
     }
 
-    (points, new_credits_observed, false)
+    StakePointsEarned {
+        points,
+        new_credits_observed,
+        force_credits_update_with_skipped_reward: false,
+    }
+}
+
+#[derive(Debug, PartialEq)]
+struct StakeRewardDistributions {
+    staker_rewards: u64,
+    voter_rewards: u64,
+    credits_observed: u64,
```

Comment:
nits: maybe better off inheriting the `new_` prefix from `StakePointsEarned` for consistency and its actual usage?